query/src/main/java/io/keen/client/java/AbsoluteTimeframe.java

```diff
@@ -46,16 +46,20 @@ public AbsoluteTimeframe(String start, String end) {
      */
     @Override
     public Map<String, Object> constructTimeframeArgs() {
-        Map timeframe = null;
+        Map<String, Object> timeframe = null;
+
         if (start != null && end != null) {
-            Map absoluteTimeframe = new HashMap<String, Object>();
+            Map<String, Object> absoluteTimeframe = new HashMap<String, Object>(2);
+
             absoluteTimeframe.put(KeenQueryConstants.START, start);
             absoluteTimeframe.put(KeenQueryConstants.END, end);
 
-            timeframe = new HashMap<String, Object>();
+            timeframe = new HashMap<String, Object>(1);
```

Comment:
Same here, I bet adding an item will cause re-allocation. Pretty minor but the purpose of specifying the initial allocation is lost.